### PR TITLE
feat(go-sdk): add the prediction database method abstractions to the sdk

### PR DESCRIPTION
**Pull request by JYisus**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other changes (ci configuration, documentation or any other kind of changes)

## Checklist:

- [ ] I have created tests for my code changes, and the tests are passing.
- [ ] I have executed the pre-commit hooks locally.
- [ ] My change requires a change to the documentation (create a new issue if the documentation has not been updated).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by kafkaphoenix — DISMISSED** [review]

File: go-sdk/sdk/prediction/store.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+)
+
+var (
+	ErrPredictionNotFound = errors.New("prediction not found")
+	ErrParsingListResult  = errors.New("error parsing prediction store search result")
+)
+
+type Filter struct {
+	RequestID string
+	Workflow  string
```
Comment (kafkaphoenix) [inline go-sdk/sdk/prediction/store.go:24]: can you order them alphabetically?

File: go-sdk/sdk/prediction/prediction.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package prediction
+
+type Prediction struct {
```
Comment (kafkaphoenix) [inline go-sdk/sdk/prediction/prediction.go:3]: this file could be also renamed to types no?

File: go-sdk/sdk/prediction/store.go (hunk 1 above)
Comment (kafkaphoenix) [reply go-sdk/sdk/prediction/store.go:24]: I think all the types should be in the same file

File: go-sdk/sdk/prediction/store.go [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+}
+
+type RedisPredictionStore struct {
+	requestID string
+	client    *redis.Client
+	metadata  *metadata.Metadata
+}
+
+func NewRedisPredictionStore(requestID string) *RedisPredictionStore {
+	opts := &redis.Options{
```
Comment (kafkaphoenix) [inline go-sdk/sdk/prediction/store.go:42]: Maybe we should also send the port right now is taking port=6379 by default, also there is a field credentials provider, is our user password coming from keycloack like minio? because in that case we need authentication again

Comment (kafkaphoenix) [reply go-sdk/sdk/prediction/store.go:42]: @dfinteligenz

**Review by alexrodfe — APPROVED** [review]

File: go-sdk/sdk/prediction/filter.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	Workflow     string
+	WorkflowType string
+	Process      string
+	Version      string
+	CreationDate TimestampRange
+}
+
+type TimestampRange struct {
+	StartDate int64
+	EndDate   int64
```
Comment (dfinteligenz) [inline go-sdk/sdk/prediction/filter.go:20]: We should expose the timestamp range as a couple of `time.Time` and do the conversion on our side, so that it's easier for the users to use the filter.

File: go-sdk/sdk/prediction/find.go [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+func (r *RedisPredictionStore) Find(ctx context.Context, filter *Filter) ([]Prediction, error) {
+	var predictions []Prediction
+
+	if err := filter.Validate(); err != nil {
+		return nil, fmt.Errorf("invalid filter: %w", err)
+	}
+
+	// set prediciton index as config variable
+	result, err := r.client.Do(ctx, "FT.SEARCH", "predictionsIdx", r.buildQueryWithFilters(filter)).Result()
```
Comment (dfinteligenz) [inline go-sdk/sdk/prediction/find.go:18]: Pending TODO

File: go-sdk/sdk/prediction/save_test.go
Comment (dfinteligenz) [inline go-sdk/sdk/prediction/save_test.go:1]: Add a couple more tests for corner cases:

- Empty payload
- Empty or invalid predictionID

Comment (dfinteligenz) [reply go-sdk/sdk/prediction/save_test.go:1]: Add missing validations, if any

**Review by dfinteligenz — CHANGES_REQUESTED** [review]

File: go-sdk/sdk/prediction/find.go [hunk 6]
```diff
@@ -14,8 +18,8 @@ func (r *RedisPredictionStore) Find(ctx context.Context, filter *Filter) ([]Pred
 		return nil, fmt.Errorf("invalid filter: %w", err)
 	}
 
-	// set prediciton index as config variable
-	result, err := r.client.Do(ctx, "FT.SEARCH", "predictionsIdx", r.buildQueryWithFilters(filter)).Result()
+	// set prediction index as config variable
```
Comment (dfinteligenz) [inline go-sdk/sdk/prediction/find.go:21]: Remove this comment
```suggestion
```

**Review by dfinteligenz — CHANGES_REQUESTED** [review]

File: go-sdk/sdk/metadata/metadata.go
[the remark this answers is not in the document]
Comment (dfinteligenz) [reply go-sdk/sdk/metadata/metadata.go:1]: Awesome, yes please, add it to the metadata @JYisus

**Review by dfinteligenz — APPROVED** [review]
